tools/docker/run.sh

```diff
@@ -6,11 +6,10 @@
 # See LICENSE file for more details.
 ###############################################################
 
-scriptdir=$(dirname "$(readlink -f "$0")")
-rootdir=$(realpath "$scriptdir/../..")
+# shellcheck source=../../tools/functions.sh
+. "$(dirname "${BASH_SOURCE[0]}")/../../tools/functions.sh"
```

Comment:
Why do you remove scriptdir? There is no reason to stop using it IMO...
```
scriptdir=$(dirname "$(readlink -f "$0")")
# shellcheck source=tools/functions.sh
. "${scriptdir}/../../tools/functions.sh"
```

Reply:
I wanted to avoid using global variables as much as possible to reduce the risk of the same variable being used in a sourced script which brought us to this mess in the first place.